pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -963,6 +966,272 @@ func (a *ApplicationBackupController) prepareResources(
+	objects []runtime.Unstructured,
+) error {
+	platformCredential, err := storkops.Instance().GetPlatformCredential(backup.Spec.PlatformCredential, backup.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if platformCredential.Spec.Type == stork_api.PlatformCredentialRancher {
+		if err = updateRancherProjects(platformCredential, backup, objects); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func updateRancherProjects(
+	platformCredential *stork_api.PlatformCredential,
+	backup *stork_api.ApplicationBackup,
+	objects []runtime.Unstructured,
+) error {
+	projects := map[string]string{}
+	for _, o := range objects {
+		metadata, err := meta.Accessor(o)
+		if err != nil {
+			return err
+		}
+		resource := o.GetObjectKind().GroupVersionKind()
+		switch resource.Kind {
+		case "Deployment", "StatefulSet", "DeploymentConfig", "IBPPeer", "IBPCA", "IBPConsole", "IBPOrderer", "ReplicaSet":
+			err := getProjectsFromPodNamespaceSelector(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "NetworkPolicy":
+			err := getProjectsFromNetworkPolicy(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolume":
+			err := getProjectsFromPV(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolumeClaim":
+			err := getProjectsFromPVC(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		}
+	}
+
+	for _, namespace := range backup.Spec.Namespaces {
+		ns, err := core.Instance().GetNamespace(namespace)
+		if err != nil {
+			return err
+		}
+		for key, val := range ns.Annotations {
+			if strings.Contains(key, utils.CattleProjectPrefix) {
+				projects[val] = ""
+			}
+		}
+		for key, val := range ns.Labels {
+			if strings.Contains(key, utils.CattleProjectPrefix) {
+				projects[val] = ""
+			}
+		}
+	}
+
+	if err := updateProjectDiplayNames(projects, platformCredential); err != nil {
+		return err
+	}
+
+	backup.Spec.RancherProjects = projects
+
+	return nil
+}
+
+// updateProjectDiplayNames updates the projectIDs with project display names
+func updateProjectDiplayNames(
+	projects map[string]string,
+	platformCredential *stork_api.PlatformCredential,
+) error {
+	rancherClient := &rancher.Rancher{}
+	rancherClient.Init(*platformCredential.Spec.RancherConfig)
+	projectList, err := rancherClient.ListProjectNames()
+	if err != nil {
+		return err
+	}
+	for key, val := range projectList {
+		if _, ok := projects[key]; ok {
+			projects[key] = val
+		}
+		data := strings.Split(key, ":")
```

Comment:
I assume this is for the case where clusterID:ProjectID is present for NS annotation right?
Rather than assigning at line#1063, can't we do the assignment either in if(len) or else case and avoid doing a delete.
When would data[1] be present in projectList?

Reply:
One more thing that I noticed on my setup is that, if we have the annotation with out clusterID:ProjectID, the ns is associated to the project. That means to associated a ns to a project, we must have the annotation with the in the format of "clusterID:ProjectID"
If this assumption is correct, we can very well avoid getting the  projectID from label. That means the project will always have x:y format.
@px-kesavan Will sync up with you and will make the code change accordingly.

Reply:
Extracted projects might have projectID with clusterID and without cluster ID . For instance let say namespace has `c-gw7q8:p-lmz7v` and namespace selector in network policy has `p-lmz7v`. Both belongs to same project and the project list has [`c-gw7q8:p-lmz7v` ,`p-lmz7v` ], instead of having both entries , the code is to combine both in a single entry (i.e) `c-gw7q8:p-lmz7v` and deleting `p-lmz7v`. So rancherproject list with always be in clusterid:projectid format. so it will be useful in project mapping , as we dont want to do for two entries , instead we can just do for one , as both belongs to same project.

Reply:
Was discussing with @siva-portworx what would be the use case where we need to store source cluster ID aka `c-gw7q8:p-lmz7v`. On destination cluster ID would be fetched by Rancher and the source cluster ID is not where used.

Reply:
As discussed, it is need to do the direct matching the restore process.